api/xds_protocol.rst

```diff
@@ -613,6 +613,27 @@ adding/removing/updating clusters. On the other hand, routes are not
 warmed, i.e., the management plane must ensure that clusters referenced
 by a route are in place, before pushing the updates for a route.
 
+.. _xds_protocol_TTL:
+
+TTL
+~~~
+
+In the event that the management server becomes unreachable, the last known configuration received by envoy will persist until the connection is reestablished. For some services, this may not be desirable. For example, in the case of a fault injection service, a management server crash at the wrong time may leave envoy in an undesirable state. The TTL setting allows envoy to remove a set of resources after a specified period of time if contact with the management server is lost. This can be used, for example, to terminate a fault injection test when the management server can no longer be reached.
+
+State of the World TTL
+^^^^^^^^^^^^^^^^^^^^^^
+
+A TTL field in the :ref:`DiscoveryResponse <envoy_api_msg_DiscoveryResponse>` specifies the TTL for all resources contained in the response. A timer is started when the response is received. When the timer expires, all resources retrieved from the management server will be removed.
+
+The TTL timer can be updated by sending a specially formed :ref:`DiscoveryResponse <envoy_api_msg_DiscoveryResponse>` with the same version as the previously applied response. The list of resources included in the response is empty and the TTL field is optional. This serves as a signal to envoy to update the TTL to the value specified or remove the TTL if none is specified.
```

Comment:
qq: did you see what Envoy does today if we send a response at the same version but w/ empty resources?

Reply:
I'm a bit worried about making the use of version and request/respond for yet another feature; we already use it for resource delivery and ack/nack. I'm a bit worried around race conditions and conflating with a real response etc.

Could we add a ttl_refresh boolean for SotW DiscoveryResponse and set this when we want to do a refresh?

Reply:
does this mean that a message with ttl_refresh = true cannot have any resources set?

Reply:
I mean, if the ttl_refresh value is set to true then resources is required to be empty and all other fields are set as they normally would be (including version_info)?

Reply:
Perhaps we should drop the 'ttl_refresh' field and simply resend the resources with a new ttl as @markdroth suggests in another part of this PR. For the use cases I we have in mind it doesn't seem like the overhead would be very high.

Reply:
For SoTW, I would have `ttl_refresh` indicate a refresh, no need for any resources. for delta xDS we would need to at least name the resources.